src/utilities/MonteCarlo/_UnitTests/test_scenarioBasicOrbitMC.py

```diff
@@ -224,8 +227,8 @@ def test_MonteCarloSimulation(show_plots):
     if show_plots:
         plt.show()
 
-    shutil.rmtree(dirName)
-    assert not os.path.exists(dirName), "No leftover data should exist after the test"
+    # shutil.rmtree(dirName)
+    tempDirectory.cleanup()
```

Comment:
It would arguably be better to use a `with` statement rather than manually calling `.cleanup()`, especially in case an exception occurs (meaning the directory would not actually get cleaned up promptly).

Of course, if any native C or C++ code causes a SIGSEGV or SIGABRT, cleanup wouldn't happen even in a `with` block`; but those signals would indicate severe bugs unto themselves!

Reply:
Made this change. Thanks for the heads up.